### DISCUSSION TITLE
Tokio Channels instead of RPC?

### DISCUSSION THREAD

**Discussion by Danconnolly** [slawlor/ractor#256] [General]
Hi, the framework looks very interesting, I'm a huge fan of the actor model and former Erlanger.

But the RPC makes me doubt using this library. Have you considered using tokio channels to handle the messaging? (https://docs.rs/tokio/latest/tokio/sync/index.html). In a custom actor implementation I'm using MPSC for sending cast messages (put the Sender of the channel in the ActorRef so it can be copied mercilessly) and oneshot channels for calls (calling function creates the channel, passes the Sender along with the call parameters).

Channels wont cross compute boundaries but isnt that a rather specialized case?

Any thoughts?

**Comment by slawlor** [comment]
RPC is backed by Tokio. But really the `rpc` module is just syntactic sugar around construction of a message with a reply channel. Especially if you're doing things like fanout, you can totally use an mpsc channel instead of a oneshot which avoids N constructions and can share the reply channel.

That's up to you how you want to go, but you can totally work around the syntactic sugar and roll your own.
